nixos/modules/services/networking/netbird.nix

```diff
@@ -1,171 +1,418 @@
+                description = "start service with the system";
+              };
+
+              hardened = mkOption {
+                type = bool;
+                default = true;
+                description = "harden the service by running as ephemeral user with minimal set of permissions";
+              };
+
+              logLevel = mkOption {
+                type = enum [
+                  # logrus loglevels
+                  "panic"
+                  "fatal"
+                  "error"
+                  "warn"
+                  "warning"
+                  "info"
+                  "debug"
+                  "trace"
+                ];
+                default = "info";
+                description = "log level of this client";
+              };
+
+              wrapper = mkOption {
+                type = package;
+                internal = true;
+                default =
+                  let
+                    exports = pipe config.environment [
+                      (mapAttrsToList (name: default:
+                        # Couldn't default with `''${${name}:${escapeShellArg default}}`
+                        # because `escapeShellArg` wraps values in single quotes,
+                        # which in turn works differenty (or rather doesn't) than double-quotes
+                        # in this shell parameter expansion
+                        ''test "''${${name}+X}" == X || export ${toShellVar name default}''))
+                      (concatStringsSep "\n")
+                    ];
+                  in
+                  pkgs.writeShellScriptBin "netbird${config.suffix}" ''
+                    ${exports}
+                    ${lib.getExe' cfg.package "netbird"} "$@"
+                  '';
               };
             };
 
-            config.environment = builtins.mapAttrs (_: mkDefault) {
-              NB_CONFIG = "/var/lib/${config.stateDir}/config.json";
-              NB_LOG_FILE = "console";
-              NB_WIREGUARD_PORT = builtins.toString config.port;
-              NB_INTERFACE_NAME = name;
-              NB_DAEMON_ADDR = "unix:///var/run/${config.stateDir}/sock";
+            config.environment = {
+              NB_CONFIG = "/var/lib/${config.name}/config.json";
+              NB_DAEMON_ADDR = "unix:///var/run/${config.name}/sock";
+              NB_INTERFACE_NAME = config.interface;
+              NB_LOG_FILE = mkOptionDefault "console";
+              NB_LOG_LEVEL = config.logLevel;
+              NB_WIREGUARD_PORT = toString config.port;
             };
           }
         )
       );
       default = { };
       description = ''
-        Attribute set of Netbird tunnels, each one will spawn a daemon listening on ...
+        Attribute set of Netbird clients, each one will spawn an daemon listening on a chosen UDP port.
+
+        A single daemon is fully operational.
+
+        Multiple daemons will interfere with each other's DNS resolution, but otherwise should remain operational.
```

Comment:
When the networks have different domains, there is no interference, the only issue is when then share the same domain

Reply:
I don't think the domain is configurable at all on the Netbird server side?

Reply:
It is when you self-host your server

Reply:
FYI:
> Hello[@nazarewk, we plan a custom DNS zone support coming next quarter. That would allow you to create custom domains for applications and peers

https://netbirdio.slack.com/archives/C02KHAE8VLZ/p1707820381270479?thread_ts=1707819591.188209&cid=C02KHAE8VLZ

Reply:
Custom DNS zone support is already implemented for self-hosted,
```sh
        ${pkgs.netbird}/bin/netbird-mgmt management \
        --dns-domain dns.domain
```
Will place every machine under dns.domain subdomain: `machine1.dns.domain`, `machine2.dns.domain`